common/src/main/java/com/gregtechceu/gtceu/common/machine/electric/FisherMachine.java

```diff
@@ -0,0 +1,368 @@
+    public static final int WATER_CHECK_SIZE = 5;
+    private static final ItemStack fishingRod = new ItemStack(Items.FISHING_ROD);
+
+    public FisherMachine(IMachineBlockEntity holder, int tier, Object... args) {
+        super(holder, tier);
+        this.inventorySize = (tier + 1) * (tier + 1);
+        this.fishingTicks = 1000 - tier * 200L;
+        this.cache = createCacheItemHandler(args);
+        this.baitHandler = createBaitItemHandler(args);
+        this.chargerInventory = createChargerItemHandler();
+        setOutputFacingItems(getFrontFacing());
+
+    }
+
+    //////////////////////////////////////
+    //*****     Initialization     *****//
+    //////////////////////////////////////
+
+    protected ItemStackTransfer createChargerItemHandler(Object... args) {
+        var transfer = new ItemStackTransfer();
+        transfer.setFilter(item -> GTCapabilityHelper.getElectricItem(item) != null);
+        return transfer;
+    }
+    @Override
+    public ManagedFieldHolder getFieldHolder() {
+        return MANAGED_FIELD_HOLDER;
+    }
+
+    protected NotifiableItemStackHandler createCacheItemHandler(Object... args) {
+        return new NotifiableItemStackHandler(this, inventorySize, IO.BOTH, IO.OUT);
+    }
+
+    protected NotifiableItemStackHandler createBaitItemHandler(Object... args) {
+        return new NotifiableItemStackHandler(this, 1, IO.BOTH, IO.BOTH);
+    }
+    @Override
+    public void onLoad() {
+        super.onLoad();
+        subscribeServerTick(this::update);
+    }
+
+    @Override
+    public boolean shouldWeatherOrTerrainExplosion() {
+        return false;
+    }
+
+    //////////////////////////////////////
+    //*********     Logic     **********//
+    //////////////////////////////////////
+
+
+
+    public void update() {
+        this.cache.exportToNearby(getOutputFacingItems());
+
+        //do not do anything if requirements are not met
+        if (this.energyContainer.getEnergyStored() < GTValues.V[getTier()] || !this.baitHandler.getStackInSlot(0).is(Items.STRING)|| this.getOffsetTimer() % this.fishingTicks != 0)
+            return;
+        int waterCount = 0;
+        int edgeSize = WATER_CHECK_SIZE;
+        for (int x = 0; x < edgeSize; x++) {
+            for (int z = 0; z < edgeSize; z++) {
+                BlockPos waterCheckPos = getPos().below().offset(x - edgeSize / 2, 0, z - edgeSize / 2);
+                if (getLevel().getBlockState(waterCheckPos).getBlock().equals(Blocks.WATER) &&
+                        getLevel().getBlockState(waterCheckPos).getMaterial() == Material.WATER) {
+                    waterCount++;
+                }
+            }
+        }
+        if(waterCount<WATER_CHECK_SIZE*WATER_CHECK_SIZE)
+            return;
+
+        LootTable lootTable = getLevel().getServer().getLootTables().get(new ResourceLocation("gameplay/fishing"));
+        FishingHook simulatedHook = new FishingHook(EntityType.FISHING_BOBBER, getLevel()) {
```

Comment:
i'd suggest creating this once in the constructor and only removing once the machine invalidates

Reply:
also, don't make a new ResourceLocation every tick.
use `BuiltInLootTables.FISHING` instead?

Reply:
Did that, also moved the lootcontext and the fishinghook to the constructor as well.